### DISCUSSION TITLE
Howto retrieve the dof_id of a node for a vector field?

### DISCUSSION THREAD

**Discussion by QuangDom** [idaholab/moose#18946] [Q&A Modules: Navier-Stokes] [answered]
Hi, I am learning Moose and I would like to know how can I get the dof_id(s) of a node for a vector field.
Why VectorMooseVariable.nodalDofIndex() returns a single value?
Thanks in advance,
/ Q.

**Comment by GiudGiud** [comment, marked as answer]
Hello

nodalDofIndex will return the index for the first component. 
You can then use componentDofIndices, defined in MooseVariableBase, to get the values for the other components.

I think the reason why is because the definition of vector variables is done through templating and that routine was never added for the vector instantiation as it doesnt make sense for the single valued instantiation.

Guillaume
